workers/go/kitchensink/kitchen_sink.go

```diff
@@ -152,8 +193,48 @@ func launchActivity(ctx workflow.Context, act *kitchensink.ExecuteActivityAction
 			RetryPolicy:            convertFromPBRetryPolicy(act.GetRetryPolicy()),
 		}
 		actCtx := workflow.WithActivityOptions(ctx, opts)
-		return workflow.ExecuteActivity(actCtx, act.ActivityType, act.Arguments).Get(actCtx, nil)
+		return withAwaitableChoice(actCtx, func(ctx workflow.Context) workflow.Future {
+			return workflow.ExecuteActivity(ctx, act.ActivityType, args)
+		}, act.GetAwaitableChoice())
+	}
+}
+
+func withAwaitableChoice(
+	ctx workflow.Context,
+	starter func(workflow.Context) workflow.Future,
+	awaitChoice *kitchensink.AwaitableChoice,
+) error {
+	cancelCtx, cancel := workflow.WithCancel(ctx)
+	fut := starter(cancelCtx)
+	var err error
+	didCancel := false
+	if awaitChoice.GetAbandon() != nil {
+		return nil
+	} else if awaitChoice.GetCancelBeforeStarted() != nil {
+		cancel()
+		didCancel = true
+		err = fut.Get(ctx, nil)
+	} else if awaitChoice.GetCancelAfterStarted() != nil {
+		_ = workflow.Sleep(ctx, 1)
+		cancel()
+		didCancel = true
+		err = fut.Get(ctx, nil)
+	} else if awaitChoice.GetCancelAfterCompleted() != nil {
+		res := fut.Get(ctx, nil)
```

Comment:
This logic seems wrong. If we are canceling after getting the result I would not expect the error to be a `CanceledError ` it should be nil.